insights/combiners/httpd_conf.py

```diff
@@ -107,78 +130,57 @@ def __init__(self, local, shared):
             self.config_data = config_files_data
 
         # Store active settings - the last parsed value us stored
-        for file_name, file_path, full_data in self.config_data:
-            for option, parsed_data in full_data.iteritems():
+        self.data = {}
+
+        def _re_wrap_data(fn, fp, pd):
+            """
+            Utility function to add `file_name` and `file_path` items to the
+            dictionary and all the dictionaries as values in it.
+
+            Args:
+                fn (str): File name
+                fp (str): File path
+                pd (dict): The changed dictionary.
+
+            """
+            for option, parsed_data in pd.iteritems():
                 if isinstance(parsed_data, dict):
-                    # For section
-                    section = option
-                    content = parsed_data
-                    if section not in self.data:
-                        self.data[section] = {}
-
-                    for k, pd in content.iteritems():
-                        values = [self.ParsedData(a.value, a.line, file_name, file_path)
-                                  for a in pd]
-                        self.data[section][k] = values
+                    _re_wrap_data(fn, fp, parsed_data)
+                elif isinstance(parsed_data, list):
+                    d = parsed_data[-1]
+                    pd[option] = self.ParsedData(d.value, d.line, fn, fp)
+
+        for file_name, file_path, full_data in self.config_data:
+            copy_data = full_data.copy()
+            _re_wrap_data(file_name, file_path, copy_data)
+            for option, parsed_data in copy_data.iteritems():
+                if option in self.data and isinstance(parsed_data, dict):
+                    self.data[option].update(parsed_data)
                 else:
-                    # For directive
-                    values = [self.ParsedData(a.value, a.line, file_name, file_path)
-                              for a in parsed_data]
-                    self.data[option] = values
+                    self.data[option] = parsed_data
 
-    def get_setting_list(self, directive, section=None):
```

Comment:
The methods 'get_setting_list' and 'get_active_setting' serve different purposes and both of them should be retained. "get_active_setting" is meant to get the last value of directive which does value overwriting, "get_setting_list" is meant for parsing either directives which does value "addition" as explained in case of HTTPD_CONF_MORE test, or for rules which warn even on currently overwritten values - in case that the config file with overwriting is removed in the future. 

Current implementation of "get_active_setting" does not get the active value of the directive so the name is misleading.

We are planing to use both in https://bugzilla.redhat.com/show_bug.cgi?id=1436224 .